Sporadic test failure (1x): GrizzlyTest

```
javax.ws.rs.ProcessingException: IOException thrown when trying to start grizzly server
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory.createHttpServer(GrizzlyHttpServerFactory.java:276)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory.createHttpServer(GrizzlyHttpServerFactory.java:110)
	at GrizzlyTest.startServer(GrizzlyTest.groovy:45)
	at io.opentelemetry.auto.test.base.HttpServerTest.setupSpec(HttpServerTest.groovy:73)
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind(Net.java:479)
	at java.base/sun.nio.ch.Net.bind(Net.java:468)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:220)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:88)
	at org.glassfish.grizzly.nio.transport.TCPNIOBindingHandler.bindToChannelAndAddress(TCPNIOBindingHandler.java:131)
	at org.glassfish.grizzly.nio.transport.TCPNIOBindingHandler.bind(TCPNIOBindingHandler.java:87)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.bind(TCPNIOTransport.java:450)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.bind(TCPNIOTransport.java:430)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.bind(TCPNIOTransport.java:421)
	at org.glassfish.grizzly.http.server.NetworkListener.start(NetworkListener.java:658)
	at org.glassfish.grizzly.http.server.HttpServer.start(HttpServer.java:264)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpServerFactory.createHttpServer(GrizzlyHttpServerFactory.java:274)
	... 3 more
```

https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-auto-instr-java/1104/workflows/5a3ed100-023e-4e3a-aeaa-8a836fe3684c/jobs/9218

--- a/testing/src/main/groovy/io/opentelemetry/auto/test/base/HttpServerTest.groovy
+++ b/testing/src/main/groovy/io/opentelemetry/auto/test/base/HttpServerTest.groovy
@@ -61,17 +61,25 @@ abstract class HttpServerTest<SERVER> extends AgentTestRunner {
   @Shared
   OkHttpClient client = OkHttpUtils.client()
   @Shared
-  int port = PortUtils.randomOpenPort()
+  int port
   @Shared
-  URI address = buildAddress()
+  URI address
+
+  def setupSpec() {
+    withRetryOnBindException({
+      setupSpecUnderRetry()
+    })
+  }
+
+  def setupSpecUnderRetry() {
+    port = PortUtils.randomOpenPort()
+    address = buildAddress()
+    server = startServer(port)
+    println getClass().name + " http server started at: http://localhost:$port/"
+  }
 
   URI buildAddress() {
     return new URI("http://localhost:$port/")
-  }
-
-  def setupSpec() {
-    server = startServer(port)
-    println getClass().name + " http server started at: http://localhost:$port/"
   }
 
   abstract SERVER startServer(int port)

